### PR TITLE
add librasterlite/1.1g

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

depends on https://github.com/conan-io/conan-center-index/pull/6799

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
@SSE4 @madebr @ericLemanissier @uilianries @prince-chrismc Call for review (I don't why some PRs are not displayed in https://github.com/prince-chrismc/conan-center-index-pending-review/issues/1).

**Comment by ericLemanissier** [comment]
side topic: It's not listed because it has no reviews and the last commit is older than 24 hours

**Comment by prince-chrismc** [comment]
We have been slacking. We were ~100 open PRs it's been creeping up all summer 🙊

File: recipes/librasterlite/all/patches/0001-fix-autotools.patch [hunk 1]
```diff
@@ -0,0 +1,27 @@
+Fix autotools:
+* Disable test and tools
+* Do not check transitive dependencies (zlib and proj)
+
+--- a/Makefile.am
++++ b/Makefile.am
+@@ -1,6 +1,6 @@
+ ACLOCAL_AMFLAGS = -I m4
+ 
+-SUBDIRS = headers lib src test
```
Comment (prince-chrismc) [inline recipes/librasterlite/all/patches/0001-fix-autotools.patch:10]: What's in the src dir?

**Review by prince-chrismc — APPROVED** [review]

**Comment by SpaceIm** [comment]
I see, criteria of https://github.com/prince-chrismc/conan-center-index-pending-review/issues/1 shouldn't reject PR due to last commit date.

File: recipes/librasterlite/all/patches/0001-fix-autotools.patch (hunk 1 above)
Comment (SpaceIm) [reply recipes/librasterlite/all/patches/0001-fix-autotools.patch:10]: utilities

File: recipes/librasterlite/all/patches/0002-nmake-honor-flags.patch [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ $(RASTERLITE_DLL):	rasterlite_i.lib
+ 
+ rasterlite_i.lib:	$(LIBOBJ)
+-	link /debug /dll /out:$(RASTERLITE_DLL) \
+-		/implib:rasterlite_i.lib $(LIBOBJ) \
+-		C:\OSGeo4W\lib\jpeg_i.lib C:\OSGeo4W\lib\libtiff_i.lib \
+-		C:\OSGeo4W\lib\libpng13.lib C:\OSGeo4W\lib\zlib.lib \
+-		C:\OSGeo4W\lib\geotiff_i.lib C:\OSGeo4W\lib\spatialite_i.lib
++	link /dll /out:$(RASTERLITE_DLL) \
++		/implib:rasterlite_i.lib $(LIBOBJ) $(SYSTEM_LIBS)
```
Comment (madebr) [inline recipes/librasterlite/all/patches/0002-nmake-honor-flags.patch:40]: In the recipe, you set `SYSTEM_LIBS` to the system libraries of the dependencies.
How are the libraries of the dependencies themselves linked? Or aren't they required?

Are the libraries of the dependencies included in `self.deps_cpp_info.system_libs`?

**Comment by prince-chrismc** [comment]
It made sense 9 months ago. But I agree ot needs to be changed

File: recipes/librasterlite/all/patches/0002-nmake-honor-flags.patch (hunk 2 above)
Comment (SpaceIm) [reply recipes/librasterlite/all/patches/0002-nmake-honor-flags.patch:40]: It's automatically handled by `VisualStudioBuildEnvironment`: https://docs.conan.io/en/latest/reference/build_helpers/visual_studio.html#visualstudiobuildenvironment

Comment (madebr) [reply recipes/librasterlite/all/patches/0002-nmake-honor-flags.patch:40]: Are you sure?
The documentation says it sets the `CL` and `LIB` environment variable.
So does it add the the libraries of the dependencies to the `CL` variable?

Shouldn't it be considered a bug that system libraries are not added?

Comment (madebr) [reply recipes/librasterlite/all/patches/0002-nmake-honor-flags.patch:40]: I checked the logs and it contains:
```
	link /dll /out:rasterlite.dll  /implib:rasterlite_i.lib lib\rasterlite.obj lib\rasterlite_gif.obj  lib\rasterlite_png.obj lib\rasterlite_jpeg.obj  lib\rasterlite_io.obj lib\rasterlite_image.obj  lib\rasterlite_tiff.obj lib\rasterlite_aux.obj  lib\rasterlite_quantize.obj lib\rasterlite_version.obj shell32.lib Ole32.lib wsock32.lib crypt32.lib ws2_32.lib advapi32.lib user32.lib
Microsoft (R) Incremental Linker Version 14.28.29333.0
Copyright (C) Microsoft Corporation.  All rights reserved.

geotiff.lib libpng16.lib spatialite.lib proj.lib freexl.lib rttopo.lib libxml2_a.lib minizip.lib tiffxx.lib tiff.lib sqlite3.lib libcurl.lib iconv.lib charset.lib geos_c.lib geos.lib bz2.lib zlib.lib libdeflatestatic.lib lzma.lib libjpeg.lib jbig.lib zstd_static.lib webpdecoder.lib webpdemux.lib webpmux.lib webp.lib libssl.lib libcrypto.lib 
   Creating library rasterlite_i.lib and object rasterlite_i.exp
```
So it looks like it does.

**Review by madebr — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**